pkg/services/accesscontrol/filter.go

```diff
@@ -84,25 +85,45 @@ func Filter(user *models.SignedInUser, sqlID, prefix string, actions ...string)
 	return SQLFilter{query.String(), ids}, nil
 }
 
-func parseScopes(prefix string, scopes []string) (ids map[int64]struct{}, hasWildcard bool) {
-	ids = make(map[int64]struct{})
+func parseScopes(prefix string, scopes []string) (ids map[interface{}]struct{}, hasWildcard bool) {
+	ids = make(map[interface{}]struct{})
+
+	rootPrefix, attributePrefix, ok := extractPrefixes(prefix)
+	if !ok {
+		return nil, false
+	}
+
+	parser := parseStringAttribute
+	if strings.HasSuffix(prefix, "id:") {
+		parser = parseIntAttribute
+	}
+
 	for _, scope := range scopes {
-		if strings.HasPrefix(scope, prefix) || scope == "*" {
-			if id := strings.TrimPrefix(scope, prefix); id == "*" || id == ":*" || id == ":id:*" {
+		if strings.HasPrefix(scope, rootPrefix) || scope == "*" {
+			if id := strings.TrimPrefix(scope, rootPrefix); id == "*" || id == strings.TrimPrefix(attributePrefix, rootPrefix)+"*" {
 				return nil, true
 			}
-			if id, err := parseScopeID(scope); err == nil {
+
+			if !ok || !strings.HasPrefix(scope, prefix) {
```

Comment:
nit: can remove `!ok` condition from here, as it's already handled before the loop.